examples/rl-ppo-continuous.rs

```diff
@@ -0,0 +1,272 @@
+        Ok(Self {
+            l1: BuildModule::try_build(device)?,
+            mu: BuildModule::try_build(device)?,
+            std: BuildModule::try_build(device)?,
+            value: BuildModule::try_build(device)?,
+        })
+    }
+}
+
+// GradientUpdate lets you update a model's parameters using gradients
+impl<const IN: usize, const INNER: usize, const OUT: usize> GradientUpdate<Cpu, f32>
+    for Network<IN, INNER, OUT>
+{
+    fn update<U>(
+        &mut self,
+        updater: &mut U,
+        unused: &mut UnusedTensors,
+    ) -> Result<(), <Cpu as HasErr>::Err>
+    where
+        U: ParamUpdater<Cpu, f32>,
+    {
+        self.l1.update(updater, unused)?;
+        self.mu.update(updater, unused)?;
+        self.std.update(updater, unused)?;
+        self.value.update(updater, unused)?;
+        Ok(())
+    }
+}
+
+// impl Module for single item
+impl<const IN: usize, const INNER: usize, const OUT: usize, T: Tape<Cpu>>
+    nn::Module<Tensor1D<IN, T>> for Network<IN, INNER, OUT>
+{
+    type Output = (Tensor1D<OUT, T>, Tensor1D<OUT, T>, Tensor1D<OUT, T>);
+
+    fn forward(&self, x: Tensor1D<IN, T>) -> Self::Output {
+        let x = self.l1.forward(x);
+        (
+            self.mu.forward(x.retaped()),
+            self.std.forward(x.retaped()),
+            self.value.forward(x),
+        )
+    }
+}
+
+// impl Module for batch of items
+impl<const BATCH: usize, const IN: usize, const INNER: usize, const OUT: usize, T: Tape<Cpu>>
+    nn::Module<Tensor2D<BATCH, IN, T>> for Network<IN, INNER, OUT>
+{
+    type Output = (
+        Tensor2D<BATCH, OUT, T>,
+        Tensor2D<BATCH, OUT, T>,
+        Tensor2D<BATCH, OUT, T>,
+    );
+
+    fn forward(&self, x: Tensor2D<BATCH, IN, T>) -> Self::Output {
+        let x = self.l1.forward(x);
+        (
+            self.mu.forward(x.retaped()),
+            self.std.forward(x.retaped()),
+            self.value.forward(x),
+        )
+    }
+}
+
+// Hyperparameters stolen from https://github.com/seungeunrho/minimalRL/blob/master/ppo-continuous.py for now
+const LEARNING_RATE: f32 = 0.0003;
+const GAMMA: f32 = 0.9;
+const LAMBDA: f32 = 0.9;
+const EPS_CLIP: f32 = 0.2;
+
+fn main() {
+    let dev: Cpu = Default::default();
+
+    // initiliaze model - all weights are 0s
+    let mut net: Network<STATE_SIZE, INNER_SIZE, ACTION_SIZE> = nn::BuildModule::build(&dev);
+
+    let mut optimizer = Adam::new(
+        &net,
+        AdamConfig {
+            lr: LEARNING_RATE,
+            betas: [0.9, 0.999],
+            eps: 1e-08,
+            weight_decay: None,
+        },
+    );
+
+    let mut state = init_state(&dev);
+
+    // run through training data
+    for _i_epoch in 0..15 {
+        let start = Instant::now();
+
+        // <>
+        let (old_log_prob, action) = {
+            // TODO: Should we avoid calculating _v? _v is the one with the original tape if that matters?
+            let (mu, std, _v) = net.forward(state.retaped());
```

Comment:
In our implementation of nn::Module, `value` is the only one of three that is not a `retaped`. Does this mean that `value` is the only one that has the complete history including `l1`? Is this then (one of) the reason for the below error about unused parameters?

Reply:
Changing to:

```rust
fn forward(&self, x: Tensor2D<BATCH, IN, T>) -> Self::Output {
    let x = self.l1.forward(x);

    let std = self.std.forward(x.retaped());
    let value = self.value.forward(x.retaped());
    (
        self.mu.forward(x),
        std,
        value
    )
}
```

Makes `UniqueId(0), UniqueId(1)` go away from the error below

Reply:
```
let loss = -surr2.minimum(surr1) + smooth_l1_loss(value, td_target, 1.0).array();
```
Using .array() to go through a plain f32 nukes the tape thing. Now using broadcast instead... Seems to resolve that error